lib/routes/utils/request_helper.js

```diff
@@ -9,9 +9,32 @@
  * @returns {boolean}
  */
 function wantsKeys (request) {
-  return request.query.keys === 'true'
+  return request.query.keys === 'true' || request.query.keys === true
+}
+
+/**
+ * Returns whether or not to perform a signin verification email.
+ * Currently, the uid is converted to a number between 0-100 which corresponds to the
+ * sign-in confirmation sample rate.
+ *
+ * @param uid
+ * @returns {boolean}
+ */
+function shouldEnableSigninConfirmation(uid, config) {
+  // Sign-in confirmation is determined by whether the feature is enabled and
+  // if the user's first two uid digits fall within the config sample rate.
+  if (config.signinConfirmation && config.signinConfirmation.enabled) {
+
+    // Returns the first 2 integers of uid
+    var uidNum = parseInt(uid.substr(0, 4), 16) % 100
```

Comment:
These uids are v4 uuids [1], and if I understand [2] correctly that means their leading bits should be uniformly random.  Definitely worth double-checking though!

[1] https://github.com/mozilla/fxa-auth-server/blob/master/lib/routes/account.js#L152
[2] https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_.28random.29